packages/x-date-pickers/src/TimeClock/Clock.tsx

```diff
@@ -316,6 +321,10 @@ export function Clock(inProps: ClockProps) {
     }
   }, [autoFocus]);
 
+  const clampValue = (newValue: number) => Math.max(minViewValue, Math.min(maxViewValue, newValue));
```

Comment:
For the `PageUp` and `PageDown` we clamp the value (the min and max act as a boundary)
For the `ArrowUp` and `ArrowDown` we circle the value (when reaching the min, we go back to the max value and vice-versa)